### PR TITLE
support Xcode 9 new export options when using sigh

### PR DESCRIPTION
**Pull request by igor-makarov**
### Checklist
- [✓] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [✓] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [✓] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [✓] I've updated the documentation if necessary.

### Motivation and Context
I've noticed @KrauseFx added support for Xcode 9 when using `match`. However, when `match` isn't used, Xcode still needs to know the provisioning profiles per bundle identifier.

### Description
I've made a small change in `gym` fastlane action that populates the `:provisoningProfiles` hash by iterating on the profiles available in `Actions.lane_context[SharedValues::SIGH_PROFILE_PATHS]`.

I've verified that this works by running this on my project, which uses `gym` to export an application containing 6 bundles, both for app store and enterprise deployment.

The change does not affect any API.

### CODE REVIEW

**Comment by igor-makarov** [comment]
I signed it!

**Comment by KrauseFx** [comment]
Hey @igor-makarov, thanks for the PR, this is awesome. I actually build something that would use the Xcode project file and fetch the correct provisioning profile from there with https://github.com/fastlane/fastlane/pull/9460. What do you think we should merge?

**Comment by igor-makarov** [comment]
I think both need to be merged. Your PR covers the logic of inferring from the project file and it's right for that. My PR covers inferring from the profiles available to `sigh`. 

For me and my build setup personally, the `sigh` inference covers one additional (and very useful) case and that is exporting from an existent archive with different profiles than originally archived - project inference will not work in this case. 

I do believe that combining both approaches would cover more use cases than each approach separately.

To make sure, I combined my changes with the one in [#9469](https://github.com/fastlane/fastlane/pull/9460) and the behavior was not contradictive. The outcome was the one I expected (profiles downloaded by `sigh` override the ones specified in the project file).

However, I did find a small bug in [#9469](https://github.com/fastlane/fastlane/pull/9460) and I will report it there.

**Comment by igor-makarov** [comment]
Following up on this PR... I'm still having issues with the other method in [#9460](https://github.com/fastlane/fastlane/pull/9460).

However, the changes proposed here are additive and pretty small-scale.

Do you think this can be merged without waiting for [#9460](https://github.com/fastlane/fastlane/pull/9460)?

File: fastlane/lib/fastlane/actions/gym.rb [hunk 1]
```diff
@@ -24,6 +24,18 @@ def self.run(values)
           # the heavy lifting for them
           values[:export_options] ||= {}
           values[:export_options][:provisioningProfiles] = Actions.lane_context[SharedValues::MATCH_PROVISIONING_PROFILE_MAPPING]
+        elsif Actions.lane_context[SharedValues::SIGH_PROFILE_PATHS]
+          # Since Xcode 9 you need to explicitly provide the provisioning profile per app target
+          # If the user used sigh we can match the profiles from sigh
+          values[:export_options] ||= {}
+          values[:export_options][:provisioningProfiles] ||= {}
+          Actions.lane_context[SharedValues::SIGH_PROFILE_PATHS].each do |profile_path|
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/gym.rb]: This looks nice, however this code might crash, therefore potentially introducing a regression to existing setups. Can you wrap the inside of the `.each` loop inside a `begin` `rescue` block similarly to what we do at https://github.com/fastlane/fastlane/pull/9460/files#diff-7b900c24bb88ea2776b2b6c4cca7e12cR98? Thanks

**Review by KrauseFx — DISMISSED** [review]

**Comment by igor-makarov** [comment]
I added `begin` `rescue` block to iteration over `sigh` profiles as requested.

**Comment by KrauseFx** [comment]
Looks great, sorry but for some reason we seem to have a conflict, do you mind resolving it? Thanks :)

**Comment by igor-makarov** [comment]
Merged!

**Comment by igor-makarov** [comment]
Whoops. I did the merging on the github website and forgot to run tests...

**Comment by igor-makarov** [comment]
Ok, conflicts resolved for real now! 😂

**Review by mpirri — APPROVED** [review]
This looks :100:, thanks for the submission!

**Comment by igor-makarov** [comment]
Excellent! Thanks for the help!

**Pull request merged**